modyn/selector/internal/selector_strategies/downsampling_strategies/scheduler_downsampling_strategy.py

```diff
@@ -0,0 +1,97 @@
+from modyn.selector.internal.selector_strategies.downsampling_strategies import AbstractDownsamplingStrategy
+from modyn.selector.internal.selector_strategies.downsampling_strategies.utils import instantiate_downsampler
+
+
+class SchedulerDownsamplingStrategy(AbstractDownsamplingStrategy):
+    def __init__(self, downsampling_config: dict, maximum_keys_in_memory: int):
+        if "first_strategy" not in downsampling_config:
```

Comment:
Each strategy is a dict (with exactly the same format of a "main" strategy)

Reply:
so right now this is hardcoded to have a first and second strategy, correct? we could also make this an array, maybe? like
```
strategies: [
{ "name": GradNorm, ..}, {"name:" Loss}, ... 
],
scheduling_interval: 3 # advance every 3 epochs until reaching the last strategy
```

Reply:
Yep, we can make also a PeriodicScheduler as you suggested. But I think that for now a threshold one is enough (I don't see any situation in which I would use a PeriodicScheduler). Or maybe I can turn this one in a list of strategies and a list of thresholds to make it more flexible.

Reply:
I think it is just a more general implementation that basically is identical and a bit less hardcoded

Reply:
Ok, makes sense. Now it's a list